py-polars/polars/utils/udfs.py

```diff
@@ -323,30 +321,22 @@ def can_attempt_rewrite(self) -> bool:
         guaranteed that using the equivalent bare constant value will return the
         same output. (Hopefully nobody is writing lambdas like that anyway...)
         """
-        if (
-            can_attempt_rewrite := self._can_attempt_rewrite.get(self._map_target, None)
-        ) is not None:
-            return can_attempt_rewrite
-        else:
-            self._can_attempt_rewrite[self._map_target] = False
-            if self._rewritten_instructions and self._param_name is not None:
-                self._can_attempt_rewrite[self._map_target] = (
-                    # check minimum number of ops, ensuring all are parseable
-                    len(self._rewritten_instructions) >= 2
-                    and all(
-                        inst.opname in OpNames.PARSEABLE_OPS
-                        for inst in self._rewritten_instructions
-                    )
-                    # exclude constructs/functions with multiple RETURN_VALUE ops
-                    and sum(
-                        1
-                        for inst in self.original_instructions
-                        if inst.opname == "RETURN_VALUE"
-                    )
-                    == 1
-                )
-
-        return self._can_attempt_rewrite[self._map_target]
+        return (
+            self._param_name is not None
+            # check minimum number of ops, ensuring all are parseable
+            and len(self._rewritten_instructions) >= 2
+            and all(
+                inst.opname in OpNames.PARSEABLE_OPS
+                for inst in self._rewritten_instructions
+            )
+            # exclude constructs/functions with multiple RETURN_VALUE ops
+            and sum(
+                1
+                for inst in self.original_instructions
+                if inst.opname == "RETURN_VALUE"
+            )
+            == 1
+        )
```

Comment:
this can all be simplified by removing the `_can_attempt_rewrite` dict

Reply:
the `BytecodeParser` class needs initialising once for each call anyway, so I'm not sure what `_can_attempt_` is supposed to be caching

Reply:
Think I've got it; this used to accelerate things when the code was more self-referential, before I split it out into more specialised classes. Seems like this came along for the ride, but I think you are right and it is no longer necessary! I say clean it up 🤔

Reply:
thanks! 

 > I think you are right and it is no longer necessary!

gentle reminder that I can't merge without approval (in case this was meant as implicit approval, no hurry though!)

Reply:
For some reason thought there was more to come, oops!
Hitting the big green button now 😅